frontend/packages/rhoas-plugin/src/components/service-list/ServiceListPage.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import ServiceInstance from './ServiceInstance';
+import NamespacedPage, {
+  NamespacedPageVariants,
+} from '@console/dev-console/src/components/NamespacedPage';
+import { history, LoadingBox } from '@console/internal/components/utils';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { useActiveNamespace } from '@console/shared';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { useTranslation } from 'react-i18next';
+import { CloudServicesRequestModel } from '../../models/rhoas';
+import { ServicesRequestCRName } from '../../const';
+import {
+  createKafkaConnection,
+  createCloudServicesRequestIfNeeded,
+  deleteKafkaConnection,
+  listOfCurrentKafkaConnectionsById,
+} from '../../utils/resourceCreators';
+import { KafkaRequest } from '../../utils/rhoas-types';
+import {
+  isResourceStatusSuccessfull,
+  isAcccesTokenSecretValid,
+  getFinishedCondition,
+} from '../../utils/conditionHandler';
+import { ServicesErrorState } from '../states/ServicesErrorState';
+
+const ServiceListPage: React.FC = () => {
+  const [currentNamespace] = useActiveNamespace();
+  const [selectedKafka, setSelectedKafka] = React.useState<number>();
+  const [currentKafkaConnections, setCurrentKafkaConnections] = React.useState<string[]>();
+  const { t } = useTranslation();
+  const [kafkaCreateError, setKafkaCreateError] = React.useState<string>();
+  const [isSubmitting, setSubmitting] = React.useState<boolean>(false);
+
+  React.useEffect(() => {
+    const createKafkaRequestFlow = async () => {
+      await createCloudServicesRequestIfNeeded(currentNamespace);
+
+      const currentKafka = await listOfCurrentKafkaConnectionsById(currentNamespace);
+      if (currentKafka) {
+        setCurrentKafkaConnections(currentKafka);
+      }
+    };
+    createKafkaRequestFlow();
+  }, [currentNamespace]);
+
+  const [watchedKafkaRequest] = useK8sWatchResource<KafkaRequest>({
+    kind: referenceForModel(CloudServicesRequestModel),
+    name: ServicesRequestCRName,
+    namespace: currentNamespace,
+    isList: false,
+    optional: true,
+  });
+
+  const remoteKafkaInstances = watchedKafkaRequest?.status?.userKafkas;
+
+  const createKafkaConnectionFlow = React.useCallback(async () => {
+    setSubmitting(true);
+    const { id, name } = remoteKafkaInstances[selectedKafka];
```

Comment:
this may break if `remoteKafkaInstances` is `undefined`? good to have check or rely on `loaded && !errMessage` of `useK8sWatchResource` above. WDYT

Reply:
At this point this never will be undefined. Also when original issue of flickering was pointed out today I have mentioned that this is temporary solution and we have plans to rewrite listPate/tables.

Reply:
Thanks for input m have felt as in line number 55 we have as check as below

```
const remoteKafkaInstances = watchedKafkaRequest?.status?.userKafkas;
```

if we can make provide default then will be safe

```
const remoteKafkaInstances = watchedKafkaRequest?.status?.userKafkas ?? { }
```